src/components/Signup/Signup.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { FormattedHTMLMessage, injectIntl } from 'react-intl';
+import { timeout } from '../../utils';
+import { Formik, Form } from 'formik';
+import { FieldGroup, InputFieldItem, CheckboxFieldItem } from '../form-helpers/form-helpers';
+
+const fieldNames = {
+  firstname: 'firstname',
+  lastname: 'lastname',
+  phone: 'phone',
+  email: 'email',
+  password: 'password',
+  accept_privacy_policies: 'accept_privacy_policies',
+  accept_promotions: 'accept_promotions',
+};
+
+/** Prepare empty values for all fields
+ * It is required because in another way, the fields are not marked as touched.
+ */
+const getFormInitialValues = () =>
+  Object.keys(fieldNames).reduce(
+    (accumulator, currentValue) => ({ ...accumulator, [currentValue]: '' }),
+    {},
+  );
+
+export default injectIntl(function({ intl }) {
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
```

Comment:
this can add in a only place and can to reuse?

Reply:
Yes, but I do not know how to do it yet.

One possible approach could be injecting `_` dependency in our _PureCI_ container. But we would couple IntlProvider with our dependency system. Since it is a single line, I prefer to keep it as it is while we think about how to reuse it.